python/ray/serve/_private/logging_utils.py

```diff
@@ -517,3 +520,155 @@ def __enter__(self):
     def __exit__(self, et, ev, tb):
         if self.level is not None:
             self.logger.setLevel(self.old_level)
+
+
+class ServeEventSummarizer:
+    """Serve-specific wrapper around Ray's EventSummarizer.
+
+    Centralizes autoscaling snapshot formatting, decision summarization,
+    throttled note emission, and change-signature calculation so controller logic
+    remains small and consistent.
+    """
+
+    def __init__(self, logger: logging.Logger):
+        self._summarizer = EventSummarizer()
+        self._logger = logger
```

Comment:
use global logger instead of passing logger into the class. This is not a pattern we have used in the repo.